presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestJoinQueries.java

```diff
@@ -33,11 +41,74 @@
 import static com.facebook.presto.tests.QueryTemplate.queryTemplate;
 import static com.google.common.collect.Iterables.getOnlyElement;
 import static java.lang.String.format;
+import static org.testng.Assert.assertNotEquals;
 import static org.testng.Assert.assertTrue;
 
 public abstract class AbstractTestJoinQueries
         extends AbstractTestQueryFramework
 {
+    @Test
+    public void testShuffledStatsWithInnerJoin()
+    {
+        // NOTE: only test shuffled stats with distributed query runner.
+        if (!(getQueryRunner() instanceof DistributedQueryRunner)) {
+            return;
+        }
+        DistributedQueryRunner queryRunner = (DistributedQueryRunner) getQueryRunner();
+
+        // Get the number of rows in orders table for query stats verification below.
```

Comment:
nit: consider hard-coding these values as they do not change; this will make the test easier to read. Alternatively, consider extracting a helper method to count number of rows in a table to avoid copy-paste.